### DISCUSSION TITLE
Improving message recipient sophistication

### DISCUSSION THREAD

**Discussion by mwpowellhtx** [CommunityToolkit/dotnet#1022] [Ideas]
The use case, something like message `A`, message `B` derived from `A`. I'd like to `IRecipient<A>`, and respond to either `A` or `B` instances, as the case may be. However, it seems I need to specificaly implement `IRecipient<B>`. How _"easy"_ (big air quotes) would it be, perhaps, to improve `IMessenger` allowing for an `A` or `B` or `Nth_degree` class hierarchy. Obviously that is the naive academic ask; by extension anything in the `B` hierarchy, for example. In broader patterns discussion, homogenously sending messages, based anywhere in a given class hierarchy, allowing for a potentially heterogeneous recipient, i.e may find receiving either an `A` or a `B` or `N` for that matter.

**Comment by echoix** [comment]
I currently use the opposite of what that idea would like to do. I use derived classes, sometimes without any changes, in order to have a different message than the base message, when the receiving function is called too often as it isn't precise enough.

So this idea would break this current behaviour and another alternative would need to be used
